core/src/types/writer.rs

```diff
@@ -35,11 +35,20 @@ use crate::*;
 /// Writer is designed to write data into given path in an asynchronous
 /// manner.
 ///
-/// # Notes
+/// ## Notes
 ///
-/// Writer is designed for appending multiple blocks which could
-/// lead to much requests. If only want to send all data in single chunk,
-/// please use [`Operator::write`] instead.
+/// Writer can be used in two ways:
```

Comment:
Is it just enough to write here? 

I think it needs also to link here from the `op.writer` as well to make the user conscious of checking here.